src/Orleans/Utils/Factory.cs

```diff
@@ -0,0 +1,36 @@
+﻿namespace Orleans
+{
+    /// <summary>
+    /// Creates an instance of <typeparamref name="TInstance"/>.
+    /// </summary>
+    /// <typeparam name="TInstance"></typeparam>
+    /// <returns></returns>
+    public delegate TInstance Factory<out TInstance>();
```

Comment:
@ReubenBond,  Understood, my original points were poorly stated as they implied that delegates could not do these things when they can, it's just less intuitive that they can or should.  As stated, I don't have a _strong_ opinion on this.  If the use of delegates is the common solution to this problem in most dependency injection systems (making the pattern familiar to most DI users), and our delegates do not conflict (name wise) with other similar patterns, I've no objections to this pattern.  Alternate preferences, but not objections :)